### PR TITLE
Make America cross-platform again (platform independent install scripts)

### PR DESCRIPTION
**Pull request by Steviey**
Motivation: Cross-platform install scripts.
Tests: Successfully tested on Win 7, but should run on any other common OS. I have no Mac :-).
Demo: npm i
Comments: 
- Done with Node.js® + ShellJs.
- Introduces a new dependency called ShellJs, providing portable Unix shell commands for Node.js®.

### CODE REVIEW

**Comment by Steviey** [comment]
The conflict is only because of cosmetic and maybe a little better performance in my own modifications. 

changed: 

1. const path = require('path'); 
to 
const sep = require('path').sep;

2. const nameWithExt = file.split('docs'+path.sep)[1];
to
const nameWithExt = file.split('docs'+sep)[1];

3. mdData[name] = fs.readFileSync('docs'+path.sep+name+'.md', {encoding: 'utf8'}); 
to
 mdData[name] = fs.readFileSync('docs'+sep+name+'.md', {encoding: 'utf8'});

4. fs.writeFileSync('website'+path.sep+'docs-dist.json', JSON.stringify(mdData));
to
fs.writeFileSync('website'+sep+'docs-dist.json', JSON.stringify(mdData));

**Comment by satya164** [comment]
btw instead of `'website'+path.sep+'docs-dist.json'`, how about doing `path.join('website', 'docs-dist.json')` ?

File: package.json [hunk 1]
```diff
@@ -55,7 +62,8 @@
     "jest": "^18.1.0",
     "react": "~15.4.0",
     "react-native": "^0.40.0",
-    "react-native-vector-icons": "^3.0.0"
+    "react-native-vector-icons": "^3.0.0",
+    "shelljs": "^0.7.6"
```
Comment (satya164) [inline package.json]: Use `yarn` to install `shelljs`

`yarn add shelljs`

It'll also update the `yarn.lock` file.

File: scripts/build-dot-flow.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+var path = require('path'), fs=require('fs'), shell = require('shelljs');
```
Comment (satya164) [inline scripts/build-dot-flow.js]: use `const` and every require on a new line

File: scripts/build-dot-flow.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const sep = require('path').sep;
+
+function fromDir(startPath,filter){
+
```
Comment (satya164) [inline scripts/build-dot-flow.js]: nit: remove new line

File: scripts/build-dot-flow.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    console.log('Starting from dir '+startPath+sep);
```
Comment (satya164) [inline scripts/build-dot-flow.js]: nit: space before and after `+`

File: scripts/build-dot-flow.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    if (!fs.existsSync(startPath)){
+        return;
+    }
+
+    var files=fs.readdirSync(startPath);
+    for(var i=0;i<files.length;i++){
```
Comment (satya164) [inline scripts/build-dot-flow.js]: spaces, and use `let` instead of `var`

File: scripts/build-dot-flow.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        var filename=path.join(startPath,files[i]);
```
Comment (satya164) [inline scripts/build-dot-flow.js]: nit: spaces around `=`

File: scripts/build-dot-flow.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        var stat = fs.lstatSync(filename);
+        var n    = filename.indexOf("__tests__");
```
Comment (satya164) [inline scripts/build-dot-flow.js]: nit: remove extra spaces

File: scripts/build-dot-flow.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        if ( (stat.isDirectory()) && (n===-1)) {
+            fromDir(filename,filter); //recurse
+        }
+        else if (filename.indexOf(filter)>=0) {
```
Comment (satya164) [inline scripts/build-dot-flow.js]: else on the same line as `}`, e.g. - `} else if (...) {`

File: scripts/build-dot-flow.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    if (!fs.existsSync(startPath)){
+        return;
+    }
+
+    var files=fs.readdirSync(startPath);
+    for(var i=0;i<files.length;i++){
+        var filename=path.join(startPath,files[i]);
+        var stat = fs.lstatSync(filename);
+        var n    = filename.indexOf("__tests__");
+        if ( (stat.isDirectory()) && (n===-1)) {
```
Comment (satya164) [inline scripts/build-dot-flow.js]: remove extra space and unnecessary parenthesis, and add space around equality operator

File: scripts/build-dot-flow.js (hunk 4 above)
Comment (satya164) [inline scripts/build-dot-flow.js]: use 2 spaces indentation

File: scripts/build-web.js [hunk 10]
```diff
@@ -0,0 +1,7 @@
+let shell = require('shelljs');
+shell.mkdir('-p', 'lib/views/');
+shell.cp('-R', 'src/views/assets', 'lib/views/assets');
+process.env.BABEL_ENV = 'publish-web';
+shell.exec('babel src -d lib');
+
```
Comment (satya164) [inline scripts/build-web.js]: remove trailing new lines

File: scripts/build-web.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+let shell = require('shelljs');
+shell.mkdir('-p', 'lib/views/');
+shell.cp('-R', 'src/views/assets', 'lib/views/assets');
+process.env.BABEL_ENV = 'publish-web';
```
Comment (satya164) [inline scripts/build-web.js]: use https://www.npmjs.com/package/cross-env

File: package.json (hunk 1 above)
Comment (satya164) [inline package.json]: Use `yarn` to install `shelljs`

`yarn add --dev shelljs`

It'll also update the `yarn.lock` file.

File: scripts/compile-docs.js [hunk 12]
```diff
@@ -16,15 +17,15 @@ function crawl(location) {
...
 
 var mdData = {};
 
 names.map(function(name) {
-  mdData[name] = fs.readFileSync('docs/'+name+'.md', {encoding: 'utf8'});
+  mdData[name] = fs.readFileSync('docs'+sep+name+'.md', {encoding: 'utf8'});
 });
 
-fs.writeFileSync('website/docs-dist.json', JSON.stringify(mdData));
+fs.writeFileSync('website'+sep+'docs-dist.json', JSON.stringify(mdData));
```
Comment (satya164) [inline scripts/compile-docs.js]: use `path.join('website', 'docs-dist.json')`

File: scripts/compile-docs.js [hunk 13]
```diff
@@ -16,15 +17,15 @@ function crawl(location) {
...
+  const nameWithExt = file.split('docs'+sep)[1];
   const name = nameWithExt.split('.md')[0];
   return name;
 });
 
 var mdData = {};
 
 names.map(function(name) {
-  mdData[name] = fs.readFileSync('docs/'+name+'.md', {encoding: 'utf8'});
+  mdData[name] = fs.readFileSync('docs'+sep+name+'.md', {encoding: 'utf8'});
```
Comment (satya164) [inline scripts/compile-docs.js]: use `path.join('docs', name + '.md')`

File: scripts/compile-docs.js [hunk 14]
```diff
@@ -16,15 +17,15 @@ function crawl(location) {
 crawl('docs');
 
 var names = files.map(function(file) {
-  const nameWithExt = file.split('docs/')[1];
+  const nameWithExt = file.split('docs'+sep)[1];
```
Comment (satya164) [inline scripts/compile-docs.js]: probably don't need to add `sep` here

File: scripts/build-rn.js [hunk 15]
```diff
@@ -0,0 +1,5 @@
+let shell = require('shelljs');
+shell.mkdir('-p', 'lib-rn/views/');
+shell.cp('-R', 'src/views/assets', 'lib-rn/views/assets');
+process.env.BABEL_ENV = 'publish-rn';
```
Comment (satya164) [inline scripts/build-rn.js]: use https://www.npmjs.com/package/cross-env

**Comment by satya164** [comment]
Please make sure ESLint passes

File: package.json [hunk 16]
```diff
@@ -13,15 +13,22 @@
     "type": "git"
   },
   "author": "Adam Miskiewicz <adam@sk3vy.com>",
+  "contributors": [
```
Comment (satya164) [inline package.json]: We don't really keep this field uptodate, so maybe remove it

**Comment by Steviey** [comment]
Unfortunately I've currently to much to do. 
Interested users may refer to my fork to submit related changes:
https://github.com/Steviey/win-react-navigation

File: scripts/build-web.js (hunk 10 above)
Comment (Steviey) [inline scripts/build-web.js]: Join is a function and as such more expensive.

**Pull request closed without merging**